Pause "Auto Refresh" when page is idle

paescuj opened the pull request:
## Scope

Pause the "Auto Refresh" functionality if the page is idle (not focused / no interaction for 5min), to avoid unnecessary requests as well a requests with expired token.

## Potential Risks / Drawbacks

None

## Review Notes / Questions

- Relevant commits by @br41nslug extracted from #17341
- Thanks to @jay-p-b-7span for initial work in #17341

---

Fixes #17083

br41nslug committed:
updated refresh logic

br41nslug committed:
fixed incorrect timer

paescuj committed:
Add changeset

paescuj committed:
Remove events when component is unmounted

rijkvanzanten commented on app/src/views/private/components/refresh-sidebar-detail.vue:
```diff
@@ -1,14 +1,15 @@
 const emit = defineEmits<{
-	(e: 'update:modelValue', value: number | null): void;
-	(e: 'refresh'): void;
+	'update:modelValue': [value: number | null];
```
If we wanna make this even nicer, we could use a

```ts
const value = defineModel();
```

and use that instead of the modelValue prop w/ emit 🙂

paescuj replied:
Good idea 🚀

paescuj committed:
Clean-up

- Use defineModel
- Move interval value check and interval clean-up inside `setRefreshInterval`
- Refresh immediately when page becomes active again, as discussed in original
  PR. Especially useful for longer intervals.
